internal/impl/nats/auth.go

```diff
@@ -26,6 +29,24 @@ func authConfToOptions(auth auth.Config, fs *service.FS) []nats.Option {
 		}
 	}
 
+	// in case a nkey file content is provided, save to a tmp file and load it
+	if auth.NKeyFileContent != "" {
+		if file, err := saveTempNkeyFile(auth.NKeyFileContent); err != nil {
+			opts = append(opts, func(*nats.Options) error { return err })
+		} else {
+			if opt, err := nats.NkeyOptionFromSeed(file); err != nil {
+				opts = append(opts, func(*nats.Options) error { return err })
+			} else {
+				opts = append(opts, opt)
+			}
+			// keep temp file for 60s and delete it
+			go func() {
+				time.Sleep(tempFileRecycleInterval)
+				deleteTempNkeyFile(file)
```

Comment:
I checked the nats client code, the options actually is stored in the client, and there is logic in the client that if failure happens, it could reconnect with the same options. So, I think the best way to handle the deletion is to delte the file when the `Close` function on the component is called.

Reply:
the problem is that we cannot transfer this file path information from auth to nats input, or we have to keep that tmp file